src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -560,17 +561,21 @@ def build_container_from_source_with_buildpack(self, image_name, source, cache_i
         except Exception as ex:
             raise ValidationError(f"Unable to run 'pack config' command to set default builder: {ex}") from ex
 
-        # If the user specifies a target port, pass it to the buildpack
-        if self.target_port:
-            command.extend(['--env', f"ORYX_RUNTIME_PORT={self.target_port}"])
-
         logger.warning("Selecting a compatible builder for the provided application source...")
         could_build_image = False
         for builder_image in builder_image_list:
             # Run 'pack build' to produce a runnable application image for the Container App
             # Specify the image as the 'build-cache' image to ensure that the local cache is used for build layers
             command = [pack_exec_path, 'build', cache_image_name, '--builder', builder_image, '--path', source, '--tag', image_name]
 
+            # If the user specifies a target port, pass it to the buildpack
+            if self.target_port:
+                command.extend(['--env', f"ORYX_RUNTIME_PORT={self.target_port}"])
```

Comment:
I'm curious about that part of the change. Does that mean that we were not setting this ORYX_RUNTIME_PORT anymore as we were reassigning command?

Reply:
It looks like the `command` variable was being overridden _after_ the target port was being set, so the target port was never being propagated to the builder. This change ensures that both the target port and telemetry environment variables are appended to the `command` variable we construct just above.

Reply:
Yes exactly. Basically, I'm surprised that we didn't hear anyone reporting that issue yet. Thanks for catching it!